frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -0,0 +1,640 @@
+          })
+        }
+      }
+      if (
+        data[index].marker !== undefined &&
+        typeof data[index].marker.color === "string"
+      ) {
+        data[index].marker = assign(data[index].marker, {
+          color: categoryColors[colorIndex % categoryColors.length],
+        })
+      }
+    })
+    colorIndex++
+  })
+
+  colorIndex = 0
+  graphIdxToCustomData.forEach(
+    (customData: Map<string, number[]>, dataIndex: number) => {
+      customData.forEach(markerIndexes => {
+        markerIndexes.forEach(markerIndex => {
+          data[dataIndex].marker.colors[markerIndex] =
+            categoryColors[colorIndex % categoryColors.length]
+        })
+        colorIndex++
+      })
+    }
+  )
+}
+
+/**
+ * This overrides the colorscale (continuous colorscale) to all graphs.
+ * @param data - spec.data
+ */
+export function applyColorscale(data: any): any {
+  const theme = useTheme()
+  data.forEach((entry: any) => {
+    entry = assign(entry, {
+      colorscale: hasLightBackgroundColor(theme)
+        ? sequentialColorscaleLightTheme
+        : sequentialColorscaleDarkTheme,
+    })
+  })
+  return data
+}
+
+/**
+ * This applies colors specifically for the table, candlestick, and waterfall plot
+ * because their dictionary structure is different from other more regular charts.
+ * @param data - spec.data
+ */
+export function applyUniqueGraphColorsData(data: any): void {
+  const theme = useTheme()
+  const { colors, genericFonts } = theme
+  data.forEach((entry: any) => {
+    // entry.type is always defined
+    if (entry.type === "table") {
+      entry.header = assign(entry.header, {
+        font: {
+          color: getGray70(theme),
+          family: genericFonts.bodyFont,
+        },
+        line: { color: colors.fadedText05, width: 1 },
+        fill: {
+          color: colors.bgMix,
+        },
+      })
+      entry.cells = assign(entry.cells, {
+        font: {
+          color: getGray90(theme),
+          family: genericFonts.bodyFont,
+        },
+        line: { color: colors.fadedText05, width: 1 },
+        fill: {
+          color: colors.bgColor,
+        },
+      })
+    } else if (entry.type === "candlestick") {
+      if (entry.decreasing === undefined) {
+        entry = assign(entry, {
+          decreasing: {
+            line: {
+              color: getDecreasingRed(theme),
+            },
+          },
+        })
+        entry = assign(entry, {
+          increasing: {
+            line: {
+              color: getIncreasingGreen(theme),
+            },
+          },
+        })
+      }
+    } else if (entry.type === "waterfall") {
+      entry = assign(entry, {
+        connector: {
+          line: {
+            color: getGray30(theme),
+            width: 2,
+          },
+        },
+      })
+      if (entry.decreasing === undefined) {
+        entry = assign(entry, {
+          decreasing: {
+            marker: {
+              color: getDecreasingRed(theme),
```

Comment:
Is decreasing red and increasing green also part of the official plotly template?  Or is this only something we define?

Reply:
https://plotly.com/python/reference/waterfall/#waterfall-decreasing